[pango] Configure error when using CMake and pkg-config

**Describe the bug**
The generated pango.pc pkg-config file references a directory that doesn't exist. This causes a CMake configuration error when linking against the pkg-config imported target.

**Environment**
- OS: Linux, Pop OS 22.04
- Compiler: gcc-11.2.0
- CMake-3.22.1

**To Reproduce**
See example repo: https://github.com/tjmatelski/pango-configure-issue
Steps to reproduce the behavior:
1. clone the above repo, `git clone --recurse-submodules https://github.com/tjmatelski/pango-configure-issue.git`
2. Init vcpkg, `cd pango-configure-issue; ./extern/vcpkg/bootstrap-vcpkg.sh`
3. Configure with CMake, `cmake -S . -B ./build -DCMAKE_TOOLCHAIN_FILE=./extern/vcpkg/scripts/buildsystems/vcpkg.cmake`
4. Get error:
```
CMake Error in CMakeLists.txt:
  Imported target "PkgConfig::PANGO_LIB" includes non-existent path

    "<path to your clone>/pango-configure-issue/extern/vcpkg/packages/pango_x64-linux/lib/pkgconfig/../../include/harfbuzz"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

**Expected behavior**
CMake configures without error.

**Additional context**
The `-I"${includedir}/harfbuzz"` in the CFlags entry in the generated pango.pc file seems to be causing the issue. Removing the following from ports/pango/portfile.cmake fixes the issue for me, however I'm not sure if this would break things for other platforms. Since harfbuzz is already listed as a dependency of pango, I would think that there is no need to have manual additions of harbuzz include flags in the pango files.

```
set(_file "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/pango.pc")
if(EXISTS "${_file}")
    vcpkg_replace_string("${_file}" [[-I"${includedir}/pango-1.0"]] [[-I"${includedir}/pango-1.0" -I"${includedir}/harfbuzz"]])
endif()
set(_file "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/pango.pc")
if(EXISTS "${_file}")
    vcpkg_replace_string("${_file}" [[-I"${includedir}/pango-1.0"]] [[-I"${includedir}/pango-1.0" -I"${includedir}/harfbuzz"]])
endif()
```

--- a/ports/atk/portfile.cmake
+++ b/ports/atk/portfile.cmake
@@ -1,27 +1,31 @@
-set(ATK_VERSION 2.38.0)
+vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 
 vcpkg_from_gitlab(
     GITLAB_URL https://gitlab.gnome.org/
     OUT_SOURCE_PATH SOURCE_PATH
     REPO GNOME/atk
-    REF "${ATK_VERSION}"
+    REF "${VERSION}"
     HEAD_REF master
     SHA512 f31951ecbdace6a18fb9f772616137cb8732163b37448fef4daf1af60ba8479c94d498dcdaf4880468c80012c77a446da585926a99704a9a940b80e546080cf3
 )
 
 vcpkg_configure_meson(
     SOURCE_PATH "${SOURCE_PATH}"
-    OPTIONS -Dintrospection=false
-    ADDITIONAL_NATIVE_BINARIES glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
-                               glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
-    ADDITIONAL_CROSS_BINARIES  glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
-                               glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
+    OPTIONS
+        -Dintrospection=false
+    ADDITIONAL_BINARIES
+        "glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'"
+        "glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'"
 )
 vcpkg_install_meson()
 
 vcpkg_copy_pdbs()
 
 vcpkg_fixup_pkgconfig()
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/atk-1.0/atk/atkmisc.h" "ifdef ATK_STATIC_COMPILATION" "if 1")
+endif()
+
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
-

--- a/ports/atk/vcpkg.json
+++ b/ports/atk/vcpkg.json
@@ -1,12 +1,20 @@
 {
   "name": "atk",
   "version": "2.38.0",
-  "port-version": 3,
+  "port-version": 4,
   "description": "GNOME Accessibility Toolkit",
   "homepage": "https://developer.gnome.org/atk/",
   "license": "GPL-2.0-only",
   "dependencies": [
     "gettext",
+    {
+      "name": "gettext",
+      "host": true,
+      "default-features": false,
+      "features": [
+        "tools"
+      ]
+    },
     "glib",
     {
       "name": "glib",

--- a/ports/graphviz/cmake-project-include.cmake
+++ b/ports/graphviz/cmake-project-include.cmake
@@ -1,0 +1,5 @@
+if(MSVC)
+    find_package(unofficial-getopt-win32 CONFIG REQUIRED)
+    set(GETOPT_LIBRARY "unofficial::getopt-win32::getopt" CACHE INTERNAL "vcpkg")
+    set(GETOPT_RUNTIME_LIBRARY "unused" CACHE INTERNAL "vcpkg")
+endif()

--- a/ports/graphviz/no-absolute-paths.patch
+++ b/ports/graphviz/no-absolute-paths.patch
@@ -1,0 +1,10 @@
+diff --git a/lib/gvc/CMakeLists.txt b/lib/gvc/CMakeLists.txt
+index b437bde..04fe8e8 100644
+--- a/lib/gvc/CMakeLists.txt
++++ b/lib/gvc/CMakeLists.txt
+@@ -1,4 +1,4 @@
+-add_definitions(-DGVC_EXPORTS -DGVLIBDIR="${LIBRARY_INSTALL_DIR}/graphviz")
++add_definitions(-DGVC_EXPORTS -DGVLIBDIR=0)
+ 
+ add_library(gvc SHARED
+   # Header files

--- a/ports/graphviz/no-absolute-paths.patch
+++ b/ports/graphviz/no-absolute-paths.patch
@@ -4,7 +4,7 @@ index b437bde..04fe8e8 100644
 +++ b/lib/gvc/CMakeLists.txt
 @@ -1,4 +1,4 @@
 -add_definitions(-DGVC_EXPORTS -DGVLIBDIR="${LIBRARY_INSTALL_DIR}/graphviz")
-+add_definitions(-DGVC_EXPORTS -DGVLIBDIR=0)
++add_definitions(-DGVC_EXPORTS -DGVLIBDIR="")
  
  add_library(gvc SHARED
    # Header files

--- a/ports/graphviz/portfile.cmake
+++ b/ports/graphviz/portfile.cmake
@@ -1,3 +1,5 @@
+set(VCPKG_POLICY_DLLS_IN_STATIC_LIBRARY enabled) # for plugins
+
 vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 
 vcpkg_from_gitlab(
@@ -9,6 +11,7 @@ vcpkg_from_gitlab(
     HEAD_REF main
     PATCHES
         fix-dependencies.patch
+        no-absolute-paths.patch
         select-plugins.patch
         static-linkage.patch
 )
@@ -98,6 +101,8 @@ endif()
 if(VCPKG_TARGET_IS_WINDOWS)
     file(GLOB plugins "${CURRENT_PACKAGES_DIR}/bin/gvplugin_*")
     file(COPY ${plugins} ${plugin_config} DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
+else()
+    file(COPY "${CURRENT_PACKAGES_DIR}/lib/graphviz" DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}")
 endif()
 
 file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/graphviz/portfile.cmake
+++ b/ports/graphviz/portfile.cmake
@@ -10,6 +10,7 @@ vcpkg_from_gitlab(
     PATCHES
         fix-dependencies.patch
         select-plugins.patch
+        static-linkage.patch
 )
 
 if(VCPKG_TARGET_IS_OSX)
@@ -51,6 +52,7 @@ vcpkg_cmake_configure(
         "-DGIT=${GIT}"
         "-DPython3_EXECUTABLE=${PYTHON3}"
         "-DPKG_CONFIG_EXECUTABLE=${CURRENT_HOST_INSTALLED_DIR}/tools/pkgconf/pkgconf"
+        "-DCMAKE_PROJECT_INCLUDE=${CMAKE_CURRENT_LIST_DIR}/cmake-project-include.cmake"
         -Dinstall_win_dependency_dlls=OFF
         -Duse_win_pre_inst_libs=OFF
         -Dwith_smyrna=OFF

--- a/ports/graphviz/static-linkage.patch
+++ b/ports/graphviz/static-linkage.patch
@@ -1,0 +1,25 @@
+diff --git a/plugin/gd/gvrender_gd.c b/plugin/gd/gvrender_gd.c
+index 960271c..2494e95 100644
+--- a/plugin/gd/gvrender_gd.c
++++ b/plugin/gd/gvrender_gd.c
+@@ -259,7 +259,7 @@ static void gdgen_missingfont(char *fontreq) {
+ #else
+ #define GD_IMPORT
+ #endif
+-GD_IMPORT extern gdFontPtr gdFontTiny, gdFontSmall, gdFontMediumBold, gdFontLarge, gdFontGiant;
++BGD_EXPORT_DATA_PROT extern gdFontPtr gdFontTiny, gdFontSmall, gdFontMediumBold, gdFontLarge, gdFontGiant;
+ 
+ void gdgen_text(gdImagePtr im, pointf spf, pointf epf, int fontcolor, double fontsize, int fontdpi, double fontangle, char *fontname, char *str)
+ {
+diff --git a/plugin/pango/CMakeLists.txt b/plugin/pango/CMakeLists.txt
+index 26749af..65d77e4 100644
+--- a/plugin/pango/CMakeLists.txt
++++ b/plugin/pango/CMakeLists.txt
+@@ -58,6 +58,7 @@ if(CAIRO_FOUND AND PANGOCAIRO_FOUND)
+   set_target_properties(gvplugin_pango PROPERTIES
+     VERSION ${GRAPHVIZ_PLUGIN_VERSION}.0.0
+     SOVERSION ${GRAPHVIZ_PLUGIN_VERSION}
++    LINKER_LANGUAGE CXX # for cairo
+   )
+ 
+   if(MINGW)

--- a/ports/graphviz/vcpkg.json
+++ b/ports/graphviz/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "graphviz",
   "version-semver": "7.0.0",
+  "port-version": 1,
   "description": "Graph Visualization Tools",
   "homepage": "https://graphviz.org/",
   "license": "EPL-1.0",

--- a/ports/gtk3/cairo-cpp-linkage.patch
+++ b/ports/gtk3/cairo-cpp-linkage.patch
@@ -1,0 +1,62 @@
+diff --git a/gtk/meson.build b/gtk/meson.build
+index 79f4f0d..33a7723 100644
+--- a/gtk/meson.build
++++ b/gtk/meson.build
+@@ -1074,6 +1074,7 @@ gtk_query_settings = executable(
+   'gtk-query-settings.c',
+   c_args: gtk_cargs,
+   dependencies: libgtk_dep,
++  link_language: 'cpp',
+   install: true
+ )
+ gtk_tools += gtk_query_settings
+@@ -1083,6 +1084,7 @@ gtk_builder_tool = executable(
+   'gtk-builder-tool.c',
+   c_args: gtk_cargs,
+   dependencies: libgtk_dep,
++  link_language: 'cpp',
+   install: true
+ )
+ gtk_tools += gtk_builder_tool
+@@ -1115,6 +1117,7 @@ gtk_update_icon_cache = executable(
+   extra_update_icon_cache_objs,
+   c_args: gtk_cargs,
+   dependencies: libgtk_dep,
++  link_language: 'cpp',
+   install: true
+ )
+ gtk_tools += gtk_update_icon_cache
+@@ -1125,6 +1128,7 @@ gtk_query_immodules = executable(
+   'gtkutils.c',
+   c_args: gtk_cargs,
+   dependencies: libgtk_dep,
++  link_language: 'cpp',
+   install: true
+ )
+ gtk_tools += gtk_query_immodules
+@@ -1134,6 +1138,7 @@ gtk_encode_symbolic_svg = executable(
+   'encodesymbolic.c',
+   c_args: gtk_cargs,
+   dependencies: libgtk_dep,
++  link_language: 'cpp',
+   install: true
+ )
+ gtk_tools += gtk_encode_symbolic_svg
+@@ -1143,6 +1148,7 @@ gtk_launch = executable(
+   'gtk-launch.c',
+   c_args: gtk_cargs,
+   dependencies: libgtk_dep,
++  link_language: 'cpp',
+   install: true
+ )
+ gtk_tools += gtk_launch
+diff --git a/meson.build b/meson.build
+index 287f0cb..d35106f 100644
+--- a/meson.build
++++ b/meson.build
+@@ -1,4 +1,4 @@
+-project('gtk+-3.0', 'c',
++project('gtk+-3.0', 'c', 'cpp',
+         version: '3.24.34',
+         default_options: [
+           'buildtype=debugoptimized',

--- a/ports/gtk3/portfile.cmake
+++ b/ports/gtk3/portfile.cmake
@@ -6,14 +6,16 @@ if(buildtrees_path_length GREATER warning_length AND CMAKE_HOST_WIN32)
     )
 endif()
 
+vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 vcpkg_from_gitlab(
     OUT_SOURCE_PATH SOURCE_PATH
     GITLAB_URL https://gitlab.gnome.org
     REPO GNOME/gtk
-    REF 3.24.34
+    REF "${VERSION}"
     SHA512 20a91e30a89070461af06b33829bc723b348806b4a785d0743af8bd4789b55dade24686e08bf1b2f0335240463aacc040134babb0605b809186b15de9cf261e4
     PATCHES
         0001-build.patch
+        cairo-cpp-linkage.patch
 )
 
 vcpkg_find_acquire_program(PKGCONFIG)
@@ -38,27 +40,25 @@ vcpkg_configure_meson(
         -Dtracker3=false            # Enable Tracker3 filechooser search
         -Dcolord=no                 # Build colord support for the CUPS printing backend
         -Dintrospection=false
-    ADDITIONAL_NATIVE_BINARIES
-        glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
-        glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
-        glib-compile-resources='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-compile-resources${VCPKG_HOST_EXECUTABLE_SUFFIX}'
-        gdbus-codegen='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/gdbus-codegen'
-        glib-compile-schemas='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-compile-schemas${VCPKG_HOST_EXECUTABLE_SUFFIX}'
-    ADDITIONAL_CROSS_BINARIES
-        glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
-        glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
-        glib-compile-resources='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-compile-resources${VCPKG_HOST_EXECUTABLE_SUFFIX}'
-        gdbus-codegen='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/gdbus-codegen'
-        glib-compile-schemas='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-compile-schemas${VCPKG_HOST_EXECUTABLE_SUFFIX}'
+    ADDITIONAL_BINARIES
+        "glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'"
+        "glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'"
+        "glib-compile-resources='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-compile-resources${VCPKG_HOST_EXECUTABLE_SUFFIX}'"
+        "gdbus-codegen='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/gdbus-codegen'"
+        "glib-compile-schemas='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-compile-schemas${VCPKG_HOST_EXECUTABLE_SUFFIX}'"
 )
 
+# Reduce command line lengths, in particular for static windows builds.
+foreach(dir IN ITEMS "${TARGET_TRIPLET}-dbg" "${TARGET_TRIPLET}-rel")
+    if(EXISTS "${CURRENT_BUILDTREES_DIR}/${dir}/build.ninja")
+        vcpkg_replace_string("${CURRENT_BUILDTREES_DIR}/${dir}/build.ninja" "/${dir}/../src/" "/src/")
+    endif()
+endforeach()
 vcpkg_install_meson()
 
 vcpkg_copy_pdbs()
 
 vcpkg_fixup_pkgconfig()
-
-file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
 
 set(GTK_TOOLS
     gtk-builder-tool
@@ -73,6 +73,4 @@ vcpkg_copy_tools(TOOL_NAMES ${GTK_TOOLS} AUTO_CLEAN)
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/etc")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
-endif()
+file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/gtk3/vcpkg.json
+++ b/ports/gtk3/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "gtk3",
   "version": "3.24.34",
+  "port-version": 1,
   "description": "Portable library for creating graphical user interfaces.",
   "homepage": "https://www.gtk.org/",
   "license": null,

--- a/ports/libgd/cmake-project-include.cmake
+++ b/ports/libgd/cmake-project-include.cmake
@@ -1,0 +1,3 @@
+if(NOT BUILD_SHARED_LIBS)
+    add_definitions(-DNONDLL)
+endif()

--- a/ports/libgd/fix-static-usage.patch
+++ b/ports/libgd/fix-static-usage.patch
@@ -1,0 +1,15 @@
+diff --git a/src/gd.h b/src/gd.h
+index 3056039..8a7608a 100644
+--- a/src/gd.h
++++ b/src/gd.h
+@@ -62,6 +62,10 @@ extern "C" {
+ #  else
+ #   define BGD_EXPORT_DATA_PROT __declspec(dllimport)
+ #  endif
++#  ifdef NONDLL
++#   undef BGD_EXPORT_DATA_PROT
++#   define BGD_EXPORT_DATA_PROT
++#  endif
+ # endif
+ # define BGD_STDCALL __stdcall
+ # define BGD_EXPORT_DATA_IMPL

--- a/ports/libgd/portfile.cmake
+++ b/ports/libgd/portfile.cmake
@@ -40,12 +40,17 @@ vcpkg_cmake_configure(
     OPTIONS
         ${FEATURE_OPTIONS}
         -DBUILD_STATIC_LIBS=${BUILD_STATIC}
+        "-DCMAKE_PROJECT_INCLUDE=${CMAKE_CURRENT_LIST_DIR}/cmake-project-include.cmake"
     OPTIONS_DEBUG
         -DENABLE_TOOLS=OFF
 )
 vcpkg_cmake_install()
 vcpkg_copy_pdbs()
 
+vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/gd.h" "ifdef BGDWIN32" "if 1") # already guarded
+if(BUILD_STATIC)
+    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/gd.h" "ifdef NONDLL" "if 1")
+endif()
 if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
     string(REPLACE "_dynamic" "" suffix "_${VCPKG_LIBRARY_LINKAGE}")
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/lib/pkgconfig/gdlib.pc" " -lgd" " -llibgd${suffix}")

--- a/ports/libgd/portfile.cmake
+++ b/ports/libgd/portfile.cmake
@@ -8,6 +8,7 @@ vcpkg_from_github(
         control-build.patch
         fix-dependencies.cmake
         fix_msvc_build.patch
+        fix-static-usage.patch
 )
 
 # Delete vendored Find modules

--- a/ports/libgd/portfile.cmake
+++ b/ports/libgd/portfile.cmake
@@ -47,7 +47,6 @@ vcpkg_cmake_configure(
 vcpkg_cmake_install()
 vcpkg_copy_pdbs()
 
-vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/gd.h" "ifdef BGDWIN32" "if 1") # already guarded
 if(BUILD_STATIC)
     vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/gd.h" "ifdef NONDLL" "if 1")
 endif()

--- a/ports/libgd/vcpkg.json
+++ b/ports/libgd/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libgd",
   "version-semver": "2.3.3",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Open source code library for the dynamic creation of images by programmers.",
   "homepage": "https://github.com/libgd/libgd",
   "license": "GD",

--- a/ports/pango/freetype2-pc.patch
+++ b/ports/pango/freetype2-pc.patch
@@ -1,0 +1,13 @@
+diff --git a/meson.build b/meson.build
+index 723e0c2..c348730 100644
+--- a/meson.build
++++ b/meson.build
+@@ -312,7 +312,7 @@ endif
+ 
+ # The first version of freetype with a pkg-config file is 2.1.5,
+ # CMake uses 'freetype' rather than 'freetype2' for the package name
+-freetype_package_name = cc.get_argument_syntax() == 'msvc' ? \
++freetype_package_name = false ? \
+                         'freetype' : 'freetype2'
+ 
+ freetype_dep = dependency(freetype_package_name,

--- a/ports/pango/portfile.cmake
+++ b/ports/pango/portfile.cmake
@@ -1,14 +1,15 @@
+vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
 vcpkg_from_gitlab(
     GITLAB_URL https://gitlab.gnome.org/
     OUT_SOURCE_PATH SOURCE_PATH
     REPO GNOME/pango
-    REF  3d3f165e77178b630d7dc2b733e64af513c99f5f #v1.50.12
-    SHA512 02ddcd0be05d6c8a84f186e6ea5c64019e34922480769f9b8e6e6883dd9d30b7d473681b6dbdf84c5a7e592f0d7bf37984adf2c9b40f37d6207b4bf40bdeaa7b
-    HEAD_REF master # branch name
+    REF "${VERSION}"
+    SHA512 30be64784fac2539741c5e085ed0cf515fdd2044e71b6c08121dfb838e50d1a31f4f9e53d9869b65aa1fa246ab5cc1bec18503a08dcf0337aece84290472e336
+    HEAD_REF master
 ) 
 
 vcpkg_configure_meson(
-    SOURCE_PATH ${SOURCE_PATH}
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         -Dintrospection=disabled # Build the GObject introspection data for Pango
         -Dfontconfig=enabled # Build with FontConfig support.
@@ -18,10 +19,9 @@ vcpkg_configure_meson(
         -Dxft=disabled # Build with xft support
         -Dfreetype=enabled # Build with freetype support
         -Dgtk_doc=false #Build API reference for Pango using GTK-Doc
-    ADDITIONAL_NATIVE_BINARIES glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
-                               glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
-    ADDITIONAL_CROSS_BINARIES  glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'
-                               glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'
+    ADDITIONAL_BINARIES
+        "glib-genmarshal='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-genmarshal'"
+        "glib-mkenums='${CURRENT_HOST_INSTALLED_DIR}/tools/glib/glib-mkenums'"
 )
 
 vcpkg_install_meson()
@@ -30,13 +30,4 @@ vcpkg_copy_pdbs()
 
 vcpkg_copy_tools(TOOL_NAMES pango-view pango-list pango-segmentation AUTO_CLEAN)
 
-file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
-
-set(_file "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig/pango.pc")
-if(EXISTS "${_file}")
-    vcpkg_replace_string("${_file}" [[-I"${includedir}/pango-1.0"]] [[-I"${includedir}/pango-1.0" -I"${includedir}/harfbuzz"]])
-endif()
-set(_file "${CURRENT_PACKAGES_DIR}/lib/pkgconfig/pango.pc")
-if(EXISTS "${_file}")
-    vcpkg_replace_string("${_file}" [[-I"${includedir}/pango-1.0"]] [[-I"${includedir}/pango-1.0" -I"${includedir}/harfbuzz"]])
-endif()
+file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/pango/portfile.cmake
+++ b/ports/pango/portfile.cmake
@@ -6,6 +6,8 @@ vcpkg_from_gitlab(
     REF "${VERSION}"
     SHA512 30be64784fac2539741c5e085ed0cf515fdd2044e71b6c08121dfb838e50d1a31f4f9e53d9869b65aa1fa246ab5cc1bec18503a08dcf0337aece84290472e336
     HEAD_REF master
+    PATCHES
+        freetype2-pc.patch
 ) 
 
 vcpkg_configure_meson(

--- a/ports/pango/vcpkg.json
+++ b/ports/pango/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "pango",
   "version": "1.50.12",
+  "port-version": 1,
   "description": "Text and font handling library.",
   "homepage": "https://ftp.gnome.org/pub/GNOME/sources/pango/",
   "license": "LGPL-2.0-only",

--- a/scripts/ci.baseline.txt
+++ b/scripts/ci.baseline.txt
@@ -784,8 +784,6 @@ opusfile:arm-uwp=fail
 opusfile:x64-uwp=fail
 paho-mqtt:arm-uwp=fail
 paho-mqtt:x64-uwp=fail
-pango:x64-windows-static=fail
-pango:x64-windows-static-md=fail
 pfring:x64-osx=fail
 pfring:arm64-osx=fail
 # pfring on Linux currently fails because its build scripts enable warnings as

--- a/versions/a-/atk.json
+++ b/versions/a-/atk.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "849d0f35a03bea3d5871f8e8bbaae5d476f20798",
+      "version": "2.38.0",
+      "port-version": 4
+    },
+    {
       "git-tree": "648ef40c1a892940c1a4c501123cafacd86e4f5a",
       "version": "2.38.0",
       "port-version": 3

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -238,7 +238,7 @@
     },
     "atk": {
       "baseline": "2.38.0",
-      "port-version": 3
+      "port-version": 4
     },
     "atkmm": {
       "baseline": "2.36.1",
@@ -2798,7 +2798,7 @@
     },
     "graphviz": {
       "baseline": "7.0.0",
-      "port-version": 0
+      "port-version": 1
     },
     "greatest": {
       "baseline": "1.5.0",
@@ -5738,7 +5738,7 @@
     },
     "pango": {
       "baseline": "1.50.12",
-      "port-version": 0
+      "port-version": 1
     },
     "pangolin": {
       "baseline": "0.8",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3814,7 +3814,7 @@
     },
     "libgd": {
       "baseline": "2.3.3",
-      "port-version": 1
+      "port-version": 2
     },
     "libgeotiff": {
       "baseline": "1.7.1",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2842,7 +2842,7 @@
     },
     "gtk3": {
       "baseline": "3.24.34",
-      "port-version": 0
+      "port-version": 1
     },
     "gtkmm": {
       "baseline": "4.6.0",

--- a/versions/g-/graphviz.json
+++ b/versions/g-/graphviz.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "8a696653e6af88aaad84721eea8ffe7aad547ea6",
+      "git-tree": "ae434295bbdf9f0f79982c5e926023188c1d575c",
       "version-semver": "7.0.0",
       "port-version": 1
     },

--- a/versions/g-/graphviz.json
+++ b/versions/g-/graphviz.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "8a696653e6af88aaad84721eea8ffe7aad547ea6",
+      "version-semver": "7.0.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "b726ae0118ef489c3588be36c8542274c1233a05",
       "version-semver": "7.0.0",
       "port-version": 0

--- a/versions/g-/graphviz.json
+++ b/versions/g-/graphviz.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "ae434295bbdf9f0f79982c5e926023188c1d575c",
+      "git-tree": "2a730e87b6af42e6fbf3af0fc7e5c4dc6785e9df",
       "version-semver": "7.0.0",
       "port-version": 1
     },

--- a/versions/g-/gtk3.json
+++ b/versions/g-/gtk3.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "dfc90d235f5a782299be10c4ce655ce95622b3a2",
+      "version": "3.24.34",
+      "port-version": 1
+    },
+    {
       "git-tree": "91d648bc4cff36c2ac289cff836ece2875566e2b",
       "version": "3.24.34",
       "port-version": 0

--- a/versions/l-/libgd.json
+++ b/versions/l-/libgd.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "842dc781d0fcbb19379a879056a31691f8f94c29",
+      "git-tree": "f08df1a1ddb937d5f782f9c84cdb375e6633dbdb",
       "version-semver": "2.3.3",
       "port-version": 2
     },

--- a/versions/l-/libgd.json
+++ b/versions/l-/libgd.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "ab2e1bc77d7f963898c62e4fabeca23d82acc5de",
+      "git-tree": "842dc781d0fcbb19379a879056a31691f8f94c29",
       "version-semver": "2.3.3",
       "port-version": 2
     },

--- a/versions/l-/libgd.json
+++ b/versions/l-/libgd.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "ab2e1bc77d7f963898c62e4fabeca23d82acc5de",
+      "version-semver": "2.3.3",
+      "port-version": 2
+    },
+    {
       "git-tree": "3198a57128bcffbbd79213ff2aa000c859483b68",
       "version-semver": "2.3.3",
       "port-version": 1

--- a/versions/p-/pango.json
+++ b/versions/p-/pango.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c6e7634fcf1696b43f8a6ec38db946b5dd279b7c",
+      "version": "1.50.12",
+      "port-version": 1
+    },
+    {
       "git-tree": "db89d2912edfafb5bb63fd34e5a774ae8f526630",
       "version": "1.50.12",
       "port-version": 0

